src/travel-aid/components/StopSignalButton.tsx

```diff
@@ -38,9 +38,7 @@ export const StopSignalButton = ({
 
   if (!isTravelAidStopButtonEnabled) return null;
 
-  const selectedCall = serviceJourney.estimatedCalls.reduce((selected, cur) =>
-    cur.quay.id === fromQuayId ? cur : selected,
-  );
+  const selectedCall = serviceJourney.estimatedCalls[fromStopPosition];
```

Comment:
Is this always correct, or could we end up with a list of estimated calls that doesn't start at 0? To me, it sounds like the more robust solution would be something like

```ts
serviceJourney.estimatedCalls.find(ec => ec.stopPositionInPattern === fromStopPosition)
```

Or am I missing something?

Reply:
I think it is always correct, but I'll just change to using `find` regardless to be safer :)